executor/api/grpc/seldon/client.go

```diff
@@ -174,5 +176,36 @@ func (s *SeldonMessageGrpcClient) Status(ctx context.Context, modelName string,
 }
 
 func (s *SeldonMessageGrpcClient) Metadata(ctx context.Context, modelName string, host string, port int32, msg payload.SeldonPayload, meta map[string][]string) (payload.SeldonPayload, error) {
-	return nil, errors.Errorf("Not implemented")
+	conn, err := s.getConnection(host, port, modelName)
+	if err != nil {
+		return s.CreateErrorPayload(err), err
+	}
+	grpcClient := proto.NewModelClient(conn)
+	resp, err := grpcClient.Metadata(grpc2.AddMetadataToOutgoingGrpcContext(ctx, meta), &empty.Empty{}, s.callOptions...)
+	if err != nil {
+		return s.CreateErrorPayload(err), err
+	}
+
+	resPayload := payload.ProtoPayload{Msg: resp}
+	return &resPayload, nil
+}
+
+func (s *SeldonMessageGrpcClient) ModelMetadata(ctx context.Context, modelName string, host string, port int32, msg payload.SeldonPayload, meta map[string][]string) (payload.ModelMetadata, error) {
```

Comment:
May be worth adding a brief descripion on top of Metadata and ModelMetadata to make it clear how each of them differ as there is still a dependency between these two that introduces some ambiguity that doesn't exist with the other routes

Reply:
If you add it as a `//` above the function it actually appears as documentation to the function which is quite useful when navigating the code